lib/plugin-api/hydrogen-kernel.js

```diff
@@ -1,15 +1,65 @@
 /* @flow */
 
 import type Kernel from "./../kernel";
-import type ZMQKernel from "./../zmq-kernel";
+
+// We define all our own types here to isolate the plugin API from internal
+// implementation details of hydrogen.
+type HydrogenResultsCallback = (
+  message: any,
+  channel: "shell" | "iopub" | "stdin"
+) => void;
+
+interface HydrogenKernelMiddlewareStack {
```

Comment:
`HydrogenKernelPlugin`?

Reply:
Is this really necessary? Why not just a class?

Reply:
Needed for flow only: https://flow.org/en/docs/types/interfaces/

It doesn't provide any functionality.